code/src/OSFramework/Grid/AbstractDataSource.ts

```diff
@@ -0,0 +1,222 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+namespace OSFramework.Grid {
+    /**
+     * Used to flatten the data array
+     * @param dataArray
+     * @returns
+     */
+    function FlattenArray(dataArray: JSON[]): JSON[] {
+        const returnDataArray = [];
+        dataArray.forEach((item) => {
+            returnDataArray.push(OSFramework.Helper.Flatten(item));
```

Comment:
Why wasn't the Flatten function moved here as well? I don't think it is used elsewhere...

Reply:
I don't move it because the ActionColumn is using it... can you tell me if it is really needed?

Reply:
At the moment yes. The Action column is using it to flatten the line data to trigger the event. Do you think that this should be done in a different way?

Reply:
FlattenArray is only used, by the DataSource, hence the reason to move it here, and make it private.
Regarding the Flatten function itself, it is used by the Action Column, so it should be in common place.